### PR TITLE
Add support for GroupConvolution in DilatedConvolutionConverter

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Support `GroupConvolution` in dilated convolution pattern*

### Tickets:
 - *101414*

### CODE REVIEW

File: src/common/transformations/tests/common_optimizations/dilated_convolution_converter.cpp [hunk 1]
```diff
@@ -112,3 +112,44 @@ TEST_F(TransformationTestsF, NegativeDilatedConvolutionConverterNonZeroPadsForNC
         manager.register_pass<pass::DilatedConvolutionConverter>();
     }
 }
+
+TEST_F(TransformationTestsF, DilatedGroupConvolutionConverter) {
+    {
+        auto data = std::make_shared<opset6::Parameter>(element::f32, Shape{1, 4, 10, 10});
+        auto filters = std::make_shared<opset6::Parameter>(element::f32, Shape{1, 4, 3, 3});
```
Comment (mvafin) [inline src/common/transformations/tests/common_optimizations/dilated_convolution_converter.cpp:119]:
```suggestion
        auto filters = std::make_shared<opset6::Parameter>(element::f32, Shape{4, 1, 1, 3, 3});
```

File: src/common/transformations/tests/common_optimizations/dilated_convolution_converter.cpp [hunk 2]
```diff
@@ -112,3 +112,44 @@ TEST_F(TransformationTestsF, NegativeDilatedConvolutionConverterNonZeroPadsForNC
...
+                                                   op::Constant::create(element::i64, Shape{4}, {1, 1, 2, 2}),
+                                                   op::Constant::create(element::i64, Shape{4}, {0, 0, 1, 1}),
+                                                   op::Constant::create(element::i64, Shape{4}, {0, 0, 1, 1}));
+        function = std::make_shared<Function>(NodeVector{batch_to_space}, ParameterVector{data, filters});
+
+        manager.register_pass<pass::DilatedConvolutionConverter>();
+    }
+    {
+        auto data = std::make_shared<opset6::Parameter>(element::f32, Shape{1, 4, 10, 10});
+        auto filters = std::make_shared<opset6::Parameter>(element::f32, Shape{1, 4, 3, 3});
```
Comment (mvafin) [inline src/common/transformations/tests/common_optimizations/dilated_convolution_converter.cpp:142]:
```suggestion
        auto filters = std::make_shared<opset6::Parameter>(element::f32, Shape{4, 1, 1, 3, 3});
```

File: src/common/transformations/src/transformations/common_optimizations/dilated_convolution_converter.cpp [hunk 3]
```diff
@@ -21,7 +22,9 @@ ov::pass::DilatedConvolutionConverter::DilatedConvolutionConverter() {
     auto pads_end_pattern = pattern::wrap_type<opset6::Constant>();
     auto space_to_batch_pattern = pattern::wrap_type<opset6::SpaceToBatch>(
         {data_pattern, block_shape_pattern, pads_begin_pattern, pads_end_pattern});
-    auto conv_pattern = pattern::wrap_type<opset6::Convolution>({space_to_batch_pattern, pattern::any_input()});
+    auto conv_p = pattern::wrap_type<opset6::Convolution>({space_to_batch_pattern, pattern::any_input()});
```
Comment (rkazants) [inline src/common/transformations/src/transformations/common_optimizations/dilated_convolution_converter.cpp:25]: can you please add comment what cases this transformation covers?

**Review by rkazants — COMMENTED** [review]
@mvafin, the changes look good to me.
Just one comment about tests that can help us to avoid such issues in the future. The current transformation test will not save us if something goes wrong prior to this transformation (FE or common part) that will generate different sub-graph so that you transformation will not be applied.
Please consider to implement the following. Create simple sub-graph model (pbtxt) from the model and pass it through FE and this transformation and make sure that this sub-graph fused to the expected one. You can use MO API for such test for example.

File: src/frontends/tensorflow/tests/convert_tricky_models.cpp [hunk 4]
```diff
@@ -221,3 +221,27 @@ TEST_F(TransformationTestsF, InjectedBodyAndIf) {
         model_ref = make_shared<Model>(OutputVector{if_op}, ParameterVector{x, y});
     }
 }
+
+TEST_F(TransformationTestsF, DISABLED_ModelWithDilatedGroupConvolution) {
```
Comment (mvafin) [inline src/frontends/tensorflow/tests/convert_tricky_models.cpp:225]:
```suggestion
// Ticket 101756
TEST_F(TransformationTestsF, DISABLED_ModelWithDilatedGroupConvolution) {
```

**Pull request merged**